### PR TITLE
Partially enable strict mode

### PR DESCRIPTION
**Pull request by BBosman**
# Pull Request

## 📖 Description

This enables TypeScript `strict` mode on a global level, but disables it (for now) on packages that don't yet conform.

It's also the first pass at making some of those packages compliant.

### 🎫 Issues

Partially implements #254

## 👩‍💻 Reviewer Notes

Did the "low hanging fruit"-packages first. Mostly casting and adding guards.

Left a TODO in `fetch-client`, for which I didn't see a quick solution.

## 📑 Test Plan

CircleCI

## ⏭ Next Steps

Do the other packages and fix the TODO in `fetch-client`.

### CODE REVIEW

File: packages/fetch-client/src/http-client.ts [hunk 1]
```diff
@@ -327,11 +328,12 @@ export class HttpClient {
...
-  const parsedHeaders = {};
-  for (const name in headers || {}) {
-    if (headers.hasOwnProperty(name)) {
-      parsedHeaders[name] = (typeof headers[name] === 'function') ? headers[name]() : headers[name];
+function parseHeaderValues(headers: Record<string, unknown> | undefined): Record<string, string>  {
+  const parsedHeaders: Record<string, string> = {};
+  const $headers = headers !== undefined ? headers : {};
+  for (const name in $headers) {
+    if ($headers.hasOwnProperty(name)) {
+      parsedHeaders[name] = (typeof $headers[name] === 'function') ? ($headers[name] as () => string)() : $headers[name] as string;
```
Comment (fkleuver) [inline packages/fetch-client/src/http-client.ts]:
```suggestion
      parsedHeaders[name] = (typeof $headers[name] === 'function')
        ? ($headers[name] as () => string)()
        : $headers[name] as string;
``` 

<a id="85cd8ef0-386e-11e9-873b-15af3aec25b1" href="none" codereview-uuid-reserved-tag />

**Comment by fkleuver** [comment]
@BBosman LGTM, just a small nit over the inline types in conditional, I think those inline types get a little hard to read at times so preferably keep them at one per line or create a new type

**Comment by BBosman** [comment]
@fkleuver Fixed. 👍

**Comment by fkleuver** [comment]
Awesome thanks!

**Pull request merged**